### ISSUE TITLE
Program crashes sometimes when pausing a download

### ISSUE THREAD

**Issue by inger147** [Bionus/imgbrd-grabber#3180]
**Steps to reproduce**

1. After doing a tag search, click Get All to queue up a download
2. On the Downloads tab, right click and select Download to start it
3. Before the download is finished, click "Pause" while it's still downloading a file.
4. The program crashes and closes.

Note: This has happened when downloading from several different booru sites, but not always, sometimes the download pauses as expected and the program doesn't crash.

**System information**

- OS: Windows 10
- Grabber version: 7.12.2

**Comment by Bionus** [comment]
Indeed, that's weird! From what I see in the source code, this must have been here for a while. I'll have a look.

**Comment by Hotwire75** [comment]
I'm using the x86_64 appimage on Debian 12 and have seen this too. I've noticed sometimes after successfully pausing and resuming a download, the next time I start a download it will have the resume button showing instead of pause, which will immediately crash it when pressed.

Also, probably technically unrelated, clicking the X to close the download window while a download's running is pretty much a guaranteed crash.
